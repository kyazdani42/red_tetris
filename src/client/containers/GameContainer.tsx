import * as React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router';
import styled from 'styled-components';

import GameController from '../components/Game/Control/GameController';
import Game from '../components/Game/Display/Game';
import LeaveRoomButton from '../components/Game/TopInfo/LeaveRoomButton';
import Spectres from '../components/Game/TopInfo/Spectres';

const BackgroundStyle = styled.div`
  background-color: #000;
  height: 100vh;
`;

const TopInfoContainer = styled.div`
  height: 10vh;
  width: 100%;
  border-bottom: 1px solid #fff;
  box-shadow: 0 10px 10px rgba(255,44,55,0.10), 0 5px 5px rgba(255,55,55,0.20);
  background-color: black;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
`;

const GameWrapper = styled.div`
  position: relative;
  height: 70vh;
  max-height: 1000px;
  width: 30%;
  min-width: 300px;
  max-width: 500px;
  margin: auto;
  margin-top: 30px;
  border: 1px solid #fff;
  box-shadow: 0 10px 10px rgba(255, 44, 55, 0.1), 0 5px 5px rgba(255, 55, 55, 0.2);
  background-color: #000;
`;

const ControlContainer = styled.div`
  height: 7vh;
  margin: auto;
  margin-top: 30px;
  width: 30%;
  min-width: 300px;
  max-width: 500px;
  box-shadow: 0 10px 10px rgba(255,44,55,0.10), 0 5px 5px rgba(255,55,55,0.20);
  background-color: #000;
  border: 1px solid #fff;
`;

interface Props {
  socket: SocketIOClient.Socket | null;
}

const GameContainer: React.SFC<Props> = ({ socket }) => {
  if (!socket) {
    return <Redirect to="/" />;
  }
  return <BackgroundStyle>
      <TopInfoContainer>
        <Spectres />
        <LeaveRoomButton />
      </TopInfoContainer>
      <GameWrapper>
        <Game />
      </GameWrapper>
      <ControlContainer>
        <GameController />
      </ControlContainer>
    </BackgroundStyle>;
};

const mapStateToProps = (state: any) => ({
  socket: state.app.socket
});

export default connect(mapStateToProps)(GameContainer);
