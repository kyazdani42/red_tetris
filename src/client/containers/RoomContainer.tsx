import * as React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import styled from 'styled-components';

import { State } from '../store';

import { Background } from '../components/Background/Background';
import CreateRoomButton from '../components/Home/CreateRoom';
import { RoomRow } from '../components/Home/RoomRow';

interface RoomsProps {
  playerName: string | null;
  rooms: RoomType[];
  socket: SocketIOClient.Socket | null;
}

const RoomWrapper = styled.div`
  position: relative;
`;

const StyledContainer = styled.div`
  font-family: ${props => props.theme.fonts.text};
  font-size: 18px;
  font-weight: 300;
  background-color: #000;
  color: #000;

  z-index: 100;
  height: 80vh;
  width: 70vw;
  margin: 10vh auto auto;
  max-width: 500px;
  border: 1px solid rgba(255,69,100,0.4);
  box-shadow: 0 10px 10px rgba(255,44,55,0.20), 0 5px 5px rgba(255,55,55,0.20);
  transition: all 1s;
  align-self: flex-end;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
`;

const RowWrapperHideOverflow = styled.div`
  height: 88%;
  width: 90%;
  overflow: hidden;
  border: 1px solid rgba(255, 255, 255, 0.1);
`;

const RowWrapper = styled.div`
  height: 100%;
  width: 100%;
  padding-right: 17px;
  overflow-y: scroll;
`;

export const RoomContainer: React.SFC<RoomsProps> = ({ rooms, socket, playerName }) => {
  if (socket) {
    const redirectUrl = `${socket.nsp}[${playerName}]`;
    return <Redirect to={redirectUrl} />;
  } else {
    return <RoomComponent rooms={rooms} />;
  }
};

export const RoomComponent: React.SFC<{ rooms: RoomsProps['rooms'] }> = ({ rooms }) => (
  <React.Fragment>
    <Background />
    <RoomWrapper>
      <StyledContainer>
        <RowWrapperHideOverflow>
          <RowWrapper>
            {getRoomRows(rooms)}
          </RowWrapper>
        </RowWrapperHideOverflow>
        <CreateRoomButton />
      </StyledContainer>
    </RoomWrapper>
  </React.Fragment>
);

export const getRoomRows = (rooms: RoomType[]): JSX.Element[] =>
  rooms.map((d, i) => <RoomRow {...d} key={`room-_${d.id}_${i}`} />);

const mapStateToProps = ({ app: { rooms, socket, playerName } }: State) => ({ rooms, socket, playerName });

export default connect(mapStateToProps)(RoomContainer);
