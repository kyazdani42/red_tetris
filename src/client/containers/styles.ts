import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body, html, #root {
    padding: 0;
    margin: 0;
  }
  @font-face {
    font-family: 'Circular Book';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Book');
  }
  @font-face {
    font-family: 'Circular Book Italic';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Book Italic');
  }
  @font-face {
    font-family: 'Circular Bold';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Bold');
  }
`;

export const BackgroundStyle = styled.div`
  background-color: #000;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
`;

export const TopInfoContainer = styled.div`
  height: 10%;
  width: 100%;
  border-bottom: 1px solid #fff;
  box-shadow: 0 10px 10px rgba(255,44,55,0.10), 0 5px 5px rgba(255,55,55,0.20);
  background-color: black;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
`;

export const GameWrapper = styled.div`
  position: relative;
  z-index: 11;
  width: 35vmin;
  height: 70vmin;
  min-width: 200px;
  min-height: 400px;
  border: 1px solid #fff;
  box-shadow: 0 10px 10px rgba(255, 44, 55, 0.1), 0 5px 5px rgba(255, 55, 55, 0.2);
  background-color: #000;
`;

export const ControlContainer = styled.div`
  height: 7%;
  width: 30%;
  margin-bottom: 20px;
  min-width: 200px;
  max-width: 400px;
  box-shadow: 0 10px 10px rgba(255,44,55,0.10), 0 5px 5px rgba(255,55,55,0.20);
  background-color: #000;
  border: 1px solid #fff;
`;

export const RoomWrapper = styled.div`
  position: relative;
`;

export const StyledContainer = styled.div`
  font-family: ${props => props.theme.fonts.text};
  font-size: 18px;
  font-weight: 300;
  background-color: #000;
  color: #000;
  z-index: 100;
  height: 80vh;
  width: 70vw;
  margin: 10vh auto auto;
  max-width: 500px;
  border: 1px solid rgba(255,69,100,0.4);
  box-shadow: 0 10px 10px rgba(255,44,55,0.20), 0 5px 5px rgba(255,55,55,0.20);
  transition: all 1s;
  align-self: flex-end;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
`;

export const RowWrapperHideOverflow = styled.div`
  height: 88%;
  width: 90%;
  overflow: hidden;
  border: 1px solid rgba(255, 255, 255, 0.1);
`;

export const RowWrapper = styled.div`
  height: 100%;
  width: 100%;
  padding-right: 17px;
  overflow-y: scroll;
`;
