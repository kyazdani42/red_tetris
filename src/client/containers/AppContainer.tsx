import * as React from 'react';
import { Route } from 'react-router-dom';
import { createGlobalStyle } from 'styled-components';

import { Background } from '../components/Background';
import GameContainer from './GameContainer';
import RoomContainer from './RoomContainer';

const GlobalStyle = createGlobalStyle`
  body, html, #root {
    width: 100%;
    height: 100%;
    padding: 0;
    margin: 0;
  }
  @font-face {
    font-family: 'Circular Book';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Book');
  }
  @font-face {
    font-family: 'Circular Book Italic';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Book Italic');
  }
  @font-face {
    font-family: 'Circular Bold';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Bold');
  }
`;

export const App = () => (
  <React.Fragment>
    <Background />
    <Route exact path="/" component={RoomContainer} />
    <Route path="/:room[:player_name]" component={GameContainer} />
    <GlobalStyle />
  </React.Fragment>
);
