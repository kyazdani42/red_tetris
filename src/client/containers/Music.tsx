import * as React from 'react';
import { connect } from 'react-redux';

import { State } from '../redux/store';

import { setMusicPlaying } from '../redux/actions';
import { ImgStyle, MusicStyle } from './styles';

interface Props {
  isPlaying: boolean;
  dispatchSetIsPlaying: () => void;
}

export const Music: React.SFC<Props> = ({ isPlaying, dispatchSetIsPlaying }) => {
  return (
    <React.Fragment>
      <MusicStyle
        onClick={() => {
          const audio: any = document.getElementById('tetris-music');
          if (audio && isPlaying) {
            audio.pause();
            dispatchSetIsPlaying();
          } else if (audio) {
            audio.play();
            dispatchSetIsPlaying();
          }
        }}
      >
        <ImgStyle src={isPlaying ? '/assets/iconpause.png' : '/assets/iconplay.png'} />
      </MusicStyle>
      <audio id="tetris-music" loop autoPlay>
        <source src="/assets/theme-tetris.mp3">
        </source>
      </audio>
    </React.Fragment>
  );
};

const mapStateToProps = (state: State) => ({
  isPlaying: state.app.musicPlaying
});

const mapDispatchToProps = (dispatch: any) => ({
  dispatchSetIsPlaying: () => dispatch(setMusicPlaying())
});

export default connect(mapStateToProps, mapDispatchToProps)(Music);
