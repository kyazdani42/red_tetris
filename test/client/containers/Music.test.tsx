// tslint:disable
import * as React from 'react';
import { shallow } from 'enzyme';

import { Music } from '../../../src/client/containers/Music';

describe('testing the component', () => {
  const props = {
    isPlaying: false,
    dispatchSetIsPlaying: () => null
  };
  const wrapper = shallow(<Music {...props} />);
  it('smoke tests the component', () => {
    expect(wrapper.update()).toMatchSnapshot();
  });
  it('checks if the source is correct', () => {
    const mus = wrapper.find('.audio-icon');
    expect(mus.prop('src')).toEqual('/assets/iconplay.png');
  })
  it('checks if the source is correct', () => {
    wrapper.setProps({ isPlaying: true })
    const mus = wrapper.find('.audio-icon');
    expect(mus.prop('src')).toEqual('/assets/iconpause.png');
  })
});

