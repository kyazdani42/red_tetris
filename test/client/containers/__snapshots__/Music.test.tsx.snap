// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the component smoke tests the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dispatchSetIsPlaying={[Function]}
    isPlaying={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <UNDEFINED
          onClick={[Function]}
        >
          <UNDEFINED
            className="audio-icon"
            src="/assets/iconplay.png"
          />
        </UNDEFINED>,
        <audio
          autoPlay={true}
          id="tetris-music"
          loop={true}
        >
          <source
            src="/assets/theme-tetris.mp3"
          />
        </audio>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <UNDEFINED
            className="audio-icon"
            src="/assets/iconplay.png"
          />,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "audio-icon",
            "src": "/assets/iconplay.png",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-dnqmqq",
              "isStatic": true,
              "rules": Array [
                "
  width: 20px;
  height: 20px;
  position: fixed;
  top: 25px;
  left: 25px;
  filter: invert(1);
",
              ],
            },
            "displayName": "styled.img",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-dnqmqq",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
  width: 30px;
  height: 30px;
  line-height: 30px;
  border-radius: 0px;
  position: fixed;
  background-color: #000;
  border: 1px solid rgba(255,255,255,0.5);
  box-shadow: 0 10px 20px rgba(255,44,55,0.30);
  top: 20px;
  left: 20px;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoPlay": true,
          "children": <source
            src="/assets/theme-tetris.mp3"
          />,
          "id": "tetris-music",
          "loop": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "/assets/theme-tetris.mp3",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        "type": "audio",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <UNDEFINED
            onClick={[Function]}
          >
            <UNDEFINED
              className="audio-icon"
              src="/assets/iconplay.png"
            />
          </UNDEFINED>,
          <audio
            autoPlay={true}
            id="tetris-music"
            loop={true}
          >
            <source
              src="/assets/theme-tetris.mp3"
            />
          </audio>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <UNDEFINED
              className="audio-icon"
              src="/assets/iconplay.png"
            />,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "audio-icon",
              "src": "/assets/iconplay.png",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-dnqmqq",
                "isStatic": true,
                "rules": Array [
                  "
  width: 20px;
  height: 20px;
  position: fixed;
  top: 25px;
  left: 25px;
  filter: invert(1);
",
                ],
              },
              "displayName": "styled.img",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-dnqmqq",
              "target": "img",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
  width: 30px;
  height: 30px;
  line-height: 30px;
  border-radius: 0px;
  position: fixed;
  background-color: #000;
  border: 1px solid rgba(255,255,255,0.5);
  box-shadow: 0 10px 20px rgba(255,44,55,0.30);
  top: 20px;
  left: 20px;
",
              ],
            },
            "displayName": "styled.button",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoPlay": true,
            "children": <source
              src="/assets/theme-tetris.mp3"
            />,
            "id": "tetris-music",
            "loop": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "/assets/theme-tetris.mp3",
            },
            "ref": null,
            "rendered": null,
            "type": "source",
          },
          "type": "audio",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
